tests/lib/external/snapd-testing-tools/remote/remote.pull

```diff
@@ -45,7 +45,7 @@ remote_pull() {
     SSH_CERT="$(_get_cert)"
 
     # shellcheck disable=SC2153,SC2086
-    $SSH_PASS scp $SSH_CERT -P "$TESTS_REMOTE_PORT" -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "$TESTS_REMOTE_USER"@"$TESTS_REMOTE_HOST":"$REMOTE_PATH" "$LOCAL_PATH"
+    $SSH_PASS scp $SSH_CERT -O -P "$TESTS_REMOTE_PORT" -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "$TESTS_REMOTE_USER"@"$TESTS_REMOTE_HOST":"$REMOTE_PATH" "$LOCAL_PATH"
```

Comment:
I'm curious on why is this is needed now...

Reply:
I've done some checks and what I see is:
* The error happens only when copying from path `/sys/firmware/efi/efivars/BootOrder-8be4df61-93ca-11d2-aa0d-00e098032b8c` directly. If that file is locally copied to, say, `$HOME`, scp on it works just fine. scp for other files works as well.
* scp 22.04 client can copy `/sys/firmware/efi/efivars/BootOrder-8be4df61-93ca-11d2-aa0d-00e098032b8c` from a UC24 image just fine
* Verbose traces do not reveal much:
```
scp: debug1: Fetching /sys/firmware/efi/efivars/BootOrder-8be4df61-93ca-11d2-aa0d-00e098032b8c to ./BootOrder-8be4df61-93ca-11d2-aa0d-00e098032b8c

BootOrder-8be4df61-93ca-11d2-aa0d-00e098032b8   0%    0     0.0KB/s   --:-- ETA
scp: debug1: truncating at 0
scp: read remote "/sys/firmware/efi/efivars/BootOrder-8be4df61-93ca-11d2-aa0d-00e098032b8c" : Failure
```

It is therefore possible that this is a scp client bug in noble. At least there is something in the attributes of the file in sysfs that it does not like, athough it is readble for our user.

Reply:
Seems definitely an sftp bug, I saw this when reproducing locally:
`Jun 12 21:18:20 qemuname sftp-server[3006]: error: process_read: seek "/sys/firmware/efi/efivars/BootOrder-8be4df61-93ca-11d2-aa0d-00e098032b8c": Illegal seek
`

Reply:
I can reproduce in 24.04, and have reported a bug: https://bugs.launchpad.net/ubuntu/+source/openssh/+bug/2069305